sky/jobs/server/core.py

```diff
@@ -149,6 +150,10 @@ def launch(
             f'{colorama.Fore.YELLOW}'
             f'Launching managed job {dag.name!r} from jobs controller...'
             f'{colorama.Style.RESET_ALL}')
+
+        # Launch with the api server's user hash, so that sky status does not
+        # show the owner of the controller as whatever user launched it first.
+        os.environ[skylet_constants.USER_ID_ENV_VAR] = utils_common.SERVER_ID
```

Comment:
Can we make this a context manager `uses_server_id_as_user_has()`? Just to make sure any other things after this won't be affected by the temporary env var setting, similar for the serve controller launching part?